### ISSUE TITLE
`no-navigation-without-resolve` should accept bare string-typed variables as external URLs

### ISSUE THREAD

**Issue by edwloef** [sveltejs/eslint-plugin-svelte#1416]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.

### What version of ESLint are you using?

9.39.1

### What version of `eslint-plugin-svelte` are you using?

3.13.0

### What did you do?

<details>
<summary>Configuration</summary>

```
import prettier from "eslint-config-prettier";
import { includeIgnoreFile } from "@eslint/compat";
import js from "@eslint/js";
import svelte from "eslint-plugin-svelte";
import globals from "globals";
import { fileURLToPath } from "node:url";
import ts from "typescript-eslint";
import svelteConfig from "./svelte.config.js";

const gitignorePath = fileURLToPath(new URL("./.gitignore", import.meta.url));

export default ts.config(
	includeIgnoreFile(gitignorePath),
	js.configs.recommended,
	...ts.configs.recommended,
	...svelte.configs.recommended,
	prettier,
	...svelte.configs.prettier,
	{
		languageOptions: {
			globals: { ...globals.browser, ...globals.node },
		},
		rules: {
			"no-undef": "off",
		},
	},
	{
		files: ["**/*.svelte", "**/*.svelte.ts", "**/*.svelte.js"],
		languageOptions: {
			parserOptions: {
				projectService: true,
				extraFileExtensions: [".svelte"],
				parser: ts.parser,
				svelteConfig,
			},
		},
		rules: {
			"svelte/prefer-const": "error",
		},
	},
	{
		rules: {
			"array-callback-return": "error",
			camelcase: "error",
			curly: "error",
			eqeqeq: ["error", "always", { null: "ignore" }],
			"no-console": "error",
			"no-duplicate-imports": "error",
			"no-nested-ternary": "error",
			"no-unneeded-ternary": "error",
			"no-var": "error",
			"object-shorthand": "error",
		},
	},
);

```
</details>

<img width="159" height="151" alt="Image" src="https://github.com/user-attachments/assets/647ac170-9f4d-468b-88e5-8fd0f94d2034" />

### What did you expect to happen?

no warning

### What actually happened?

`Unexpected href link without resolve().`

### Link to **GitHub Repo** with Minimal Reproducible Example

not currently public :(

**Comment by edwloef** [comment]
Here's a reproducer:
```svelte
<script lang="ts">
	let { value }: { value: string } = $props();
</script>

<a href={value}></a>
```

**Comment by marekdedic** [comment]
Hi, thanks for the issue, but this seems to basically defeat the whole purpose of the rule? How is this different from just turning the rule off?

**Comment by edwloef** [comment]
As far as I can tell the rule is meant to provide type-safety for internal navigation, since you'd get type errors for relative URLs that don't actually exist. However, if you e.g. get external URLs from some API and want to display them to the user, the lint would false-positive. Since I don't think it'd be common for relative URLs to be present as just a bare string variable, to me it'd make sense to assume this case to fall under the exception of "`<a>` links to absolute URLs".

**Comment by WayneMorganUK** [comment]
I have created a repro here: [ https://github.com/WayneMorganUK/svelte-url-tester]( https://github.com/WayneMorganUK/svelte-url-tester)
In vs code you will need to run `pnpm lint` to see the errors, in zed the errors are highlighted in the ide. 
When the url is passed as a prop it shows a linting error. It does not show an error if the url is referenced directly or declared in the same component. Image below is from Zed editor

<img width="773" height="413" alt="Image" src="https://github.com/user-attachments/assets/12e76fc3-e48d-4a42-bce3-dbda759f9cd4" />

**Comment by marekdedic** [comment]
Thanks. The `newUrl` example makes sense. However, how do you know that `url` is an absolute URL? There's no way to tell...

**Comment by WayneMorganUK** [comment]
The problem is that its treating every link as an internal link.  Even if I add `rel="noopener noreferrer" target="_blank"` its still treating it as an internal link. Surely, if any of these are present we can treat it as an external link.

**Comment by sacrosanctic** [comment]
https://github.com/sveltejs/kit/issues/647#issuecomment-3549554959

Related workaround

**Comment by marekdedic** [comment]
It should now work with `<a rel="external">`, the rest will follow.

**Comment by sacrosanctic** [comment]
This can be closed

**Comment by marekdedic** [comment]
Yes, but we should open a new issue for the `URL.href` example by @WayneMorganUK :)

**Comment by sacrosanctic** [comment]
@marekdedic I actually have a bigger version of that bug. Even `<a href="/foo">foo</a>` does not show up on vsc, i must run `npm run lint`. Is it just me?

**Comment by marekdedic** [comment]
@sacrosanctic I don't use VSCode, so I can't help you with that. If `npx eslint` works, than the problem is probably not in this package.

**Comment by sacrosanctic** [comment]
Ya, I'll look around in the vscode extension repo.

**Comment by marekdedic** [comment]
@WayneMorganUK I opened a separate issue for `URL`: #1487
